ext/standard/array.c

```diff
@@ -4347,8 +4347,8 @@ PHP_FUNCTION(array_pad)
 	input_size = zend_hash_num_elements(Z_ARRVAL_P(input));
 	pad_size_abs = ZEND_ABS(pad_size);
 	if (pad_size_abs < 0 || pad_size_abs - input_size > Z_L(1048576)) {
-		php_error_docref(NULL, E_WARNING, "You may only pad up to 1048576 elements at a time");
-		RETURN_FALSE;
+		zend_throw_error(NULL, "Only 1048576 elements may be pad up at a time.");
```

Comment:
I think the original error-text should be kept

Reply:
Agree that the previous error message should be kept. As a general note: Exception messages should not be terminated by a period.